src/textual/command.py

```diff
@@ -735,6 +737,8 @@ def _show_no_matches() -> None:
                 # discovered is a situation we don't need to highlight.
                 self._list_visible = False
 
+        self._hit_count = 0
```

Comment:
Minor but `_start_no_matches_countdown` seems like a strange place for this (seems out of scope). I'd put it inside `gather_commands` where we check `if option_list.options_count == 0`. Take it or leave it 😄